components/tooltip/tooltip-help.js

```diff
@@ -0,0 +1,92 @@
+import '../colors/colors.js';
+import '../tooltip/tooltip.js';
+import { css, html, LitElement } from 'lit';
+import { bodySmallStyles } from '../typography/styles.js';
+import { FocusMixin } from '../../mixins/focus-mixin.js';
+import { FocusVisiblePolyfillMixin } from '../../mixins/focus-visible-polyfill-mixin.js';
+
+/**
+ * A component used to display additional information when users focus or hover on certain text.
+ * @slot - Default content placed inside of the tooltip
+ * @fires d2l-tooltip-show - Dispatched when the tooltip is opened
+ * @fires d2l-tooltip-hide - Dispatched when the tooltip is closed
+ */
+class HelpTooltip extends FocusMixin(FocusVisiblePolyfillMixin(LitElement)) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: The text that will be rendered as the Help Tooltip component.
+			 * @type {String}
+			 */
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [bodySmallStyles, css`
+			:host {
+				display: inline-block;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			#d2l-tooltip-help-text {
+				text-decoration-color: var(--d2l-color-galena);
+				text-decoration-line: underline;
+				text-decoration-style: dashed;
+				text-decoration-thickness: 1px;
+				text-underline-offset: 0.15rem;
+			}
+			#d2l-tooltip-help-text:focus {
+				outline-style: none;
+			}
+			#d2l-tooltip-help-text.focus-visible {
+				border-radius: 0.025rem;
+				outline: 2px solid var(--d2l-color-celestine);
+				outline-offset: 0.15rem;
+				text-underline-offset: 0.05rem;
+			}
+		`];
+	}
+
+	static get focusElementSelector() {
+		return 'span';
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.text || this.text.length === 0) {
+			console.warn('Help Tooltip component requires text.');
+		}
+	}
+
+	render() {
+		return html`
+			<span 
+				id="d2l-tooltip-help-text" 
```

Comment:
Is there a reason to use `getUniqueId` instead of specifying an id?

What do you mean by `space`?

Reply:
So in general you wouldn't want duplicate ids on a page since a selector for that id would just grab the first occurrence. In this case, functionally I believe it would work either way since it would be getting used in the shadow root. To me personally it still seems like good practice to use unique ids when using ids (e.g., someone could copy/paste this code but actually have potential duplicate id issues), but others may differ on this.

By space I meant `span`🤦‍♀️

Reply:
I wouldn't worry about using `getUniqueId` here -- we used to avoid them when we needed to handle browsers that didn't support shadow DOM, but we're thankfully past that now.

Reply:
I'll keep the Id as is then